### PR TITLE
CLI command to list currently loaded pods from runtime

### LINKED ISSUE #333: Add `spice pods list` to CLI
List currently loaded pods from the runtime.

```bash
% spice pods list
```

CLI calls `http://localhost:8000/api/v0.1/pods` -> JSON [{
    "name": "trader",
     "manifest_path": "/Users/luke/trader/spicepods/trader.yaml"
}]

Print to command line:

```bash
NAME         MANIFEST_PATH
trader         /Users/luke/trader/spicepods/trader.yaml
```
*Other PRs*
- [x] Documentation

spice-pods-list

### PR DESCRIPTION
**Pull request by sboorlagadda**
1) spice shows pods as a command

```bash
❯ ./spice
Spice.ai CLI

Usage:
  spice [command]

Available Commands:
  action      Maintains actions
  add         Add Pod - adds a pod to the project
  completion  generate the autocompletion script for the specified shell
  export      Export Pod - export a pod
  help        Help about any command
  import      Import Pod - import a pod
  init        Initialize Pod - initializes a new pod in the project
  pods        Retrieve pods
  reward      Maintains rewards
  run         Run Spice.ai - starts the Spice.ai runtime, installing if necessary
  train       Train Pod - start a pod training run
  upgrade     Upgrades Spice.ai runtime if necessary
  version     Spice CLI version

Flags:
  -h, --help   help for spice

Use "spice [command] --help" for more information about a command.
```
2) `spice pods -h` shows a help

```bash
❯ ./spice pods -h
Retrieve pods

Usage:
  spice pods [command]

Aliases:
  pods, pods

Examples:

spice pods list


Available Commands:
  list        Lists available pods in the runtime

Flags:
  -h, --help   Prints this help message

Use "spice pods [command] --help" for more information about a command.

```
3) `spice pods list -h` shows a help
```bash
❯ ./spice pods list -h
Lists available pods in the runtime

Usage:
  spice pods list [flags]

Examples:

spice pods list


Flags:
  -h, --help   Prints this help message
  
```
4) This is how the command lists the output
```bash  
  ❯ ./spice pods list
  NAME              MANIFEST PATH
  customprocessor   /userapp/spicepods/customprocessor.yaml
  event-categories  /userapp/spicepods/event-categories.yaml
  event-tags        /userapp/spicepods/event-tags.yaml
  trader            /userapp/spicepods/trader.yaml
```

5) if the runtime not running
```bash
❯ ./spice pods list
failed to reach http://localhost:8000. is the spice runtime running?
```

Closes #333

### CODE REVIEW

File: pkg/cli/cli_test.go [hunk 1]
```diff
@@ -99,3 +100,13 @@ func testRewardsAddCmd(root *cobra.Command) func(*testing.T) {
 		assert.Contains(t, pod.Rewards(), "jump")
 	}
 }
+
+// Tests pod list lists all availalbe pods
+func testPodListCmd(root *cobra.Command) func(*testing.T) {
+	return func(t *testing.T) {
+		output, err := executeCommand(root, "pods", "list")
```
Comment (sboorlagadda) [inline pkg/cli/cli_test.go]: @lukekim @phillipleblanc How can I add pods so I can add a test?

File: pkg/cli/cli_test.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ func TestCLI(t *testing.T) {
 	t.Run("initCmd() - spice init foo creates a skeleton pod", testInit(cmd.RootCmd))
 	t.Run("actionAddCmd() - spice action add jump adds an action", testActionAddCmd(cmd.RootCmd))
 	t.Run("rewardAddCmd() - spice reward add adds default rewards", testRewardsAddCmd(cmd.RootCmd))
+	t.Run("podListCmd() - spice pod list list available pods", testPodListCmd(cmd.RootCmd))
```
Comment (lukekim) [inline pkg/cli/cli_test.go]:
```suggestion
	t.Run("podListCmd() - spice pod list lists loaded pods", testPodListCmd(cmd.RootCmd))
```

File: pkg/cli/cli_test.go [hunk 3]
```diff
@@ -99,3 +100,13 @@ func testRewardsAddCmd(root *cobra.Command) func(*testing.T) {
 		assert.Contains(t, pod.Rewards(), "jump")
 	}
 }
+
+// Tests pod list lists all availalbe pods
```
Comment (lukekim) [inline pkg/cli/cli_test.go]:
```suggestion
// Tests pod list lists currently loaded pods
```

File: pkg/cli/cli_test.go (hunk 1 above)
Comment (lukekim) [reply pkg/cli/cli_test.go]: This should probably actually be part of the e2e tests under `test/e2e` as it tests CLI + Runtime interaction. You can see examples in that, where the runtime and pods are loaded.

Comment (lukekim) [reply pkg/cli/cli_test.go]: If you wanted to also add a test here, you would need to mock the HTTP call to the runtime.

Comment (sboorlagadda) [reply pkg/cli/cli_test.go]: I probably can `run` first in `spicepods/foo.yaml`

File: pkg/cli/cmd/pods.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+			fmt.Printf("failed to get available pods from runtime: %s\n", err.Error())
+			return
+		}
+
+		defer response.Body.Close()
+		w := new(tabwriter.Writer)
+		w.Init(os.Stdout, 8, 8, 0, '\t', 0)
+		defer w.Flush()
+
+		fmt.Fprintf(w, "%s\t%s\n", "Pod Name", "Manifest")
```
Comment (lukekim) [inline pkg/cli/cmd/pods.go]: I'd replace this with `"github.com/gocarina/gocsv"` + `"github.com/olekukonko/tablewriter"`. Then you can handle variable widths with `gocsv.MarshalString`. Check out an example here: https://github.com/dapr/cli/blob/f92eebacc6c5530e326e96b9f6bc49f7cafff58b/utils/utils.go#L35

**Review by lukekim — CHANGES_REQUESTED** [review]
Awesome start, thanks @sboorlagadda !

File: test/e2e/cli.go [hunk 5]
```diff
@@ -37,3 +37,8 @@ func (c *cli) startCliCmd(args ...string) (*exec.Cmd, error) {
 
 	return cmd, nil
 }
+
+func (c *cli) runCliCmdOutput(args ...string) ([]byte, error) {
```
Comment (sboorlagadda) [inline test/e2e/cli.go:41]: @lukekim Do you suggest a better name for the func?

**Review by phillipleblanc — APPROVED** [review]

**Review by lukekim — APPROVED** [review]

File: test/e2e/cli.go (hunk 5 above)
Comment (lukekim) [reply test/e2e/cli.go:41]: Fine for now!

**Pull request merged**